core/spec/models/spree/order_cancellations/cancelled_items_adjuster_spec.rb

```diff
@@ -0,0 +1,24 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe Spree::OrderCancellations::CancelledItemsAdjuster do
+  let(:order) { create(:order_ready_to_ship) }
+  subject { described_class.new(order).adjust! }
+
+  it "does not change anything on an order without canceled items" do
+    expect { subject }.not_to change { order.line_items.sum(&:total_before_tax) }
+  end
+
+  context "if there is a canceled item" do
+    before do
+      Spree::OrderCancellations.new(order).short_ship([order.inventory_units.first])
+      # manually update the value of the cancellation adjustment
+      order.line_items.first.adjustments.detect(&:cancellation?).update(amount: -2)
+    end
+
+    it "changes the line item's total before tax" do
+      expect { subject }.to change { order.reload.line_items.sum(&:total_before_tax) }
```

Comment:
Would love if we can assert what the amount is changed from/to as this could still pass if we make changes to this class, while doing the wrong thing. At least having the amounts there will surface any unintended changes to the behaviour.

Reply:
It's a bit problematic because I'm essentially changing the adjustment amount in the `before` block to something random, and then it corrects it back. I can add the amounts, but really only the `to` dimension is interesting.

Reply:
That makes sense, setting up realistic taxes which get recalculated correctly in specs is a pain 😩